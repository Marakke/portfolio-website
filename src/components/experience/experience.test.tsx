import React from "react";
import { render } from "@testing-library/react";
import * as Gatsby from "gatsby";
import Experience from "./index";

const useStaticQuery = jest.spyOn(Gatsby, `useStaticQuery`);

const mockUseStaticQuery = {
  experience: {
    education: {
      title: "Education Title",
      university: "Test University",
      programmes: [
        {
          name: "Programme",
          startYear: "2020",
          endYear: "2023"
        }
      ]
    },
    workExperience: {
      title: "Work Experience Title",
      workPlaces: [
        {
          name: "Work Place",
          startYear: "2020",
          endYear: "2023"
        }
      ]
    }
  }
};

describe("Experience", () => {
  beforeEach(() => {
    useStaticQuery.mockImplementation(() => mockUseStaticQuery);
  });

  afterEach(() => {
    jest.restoreAllMocks();
  });

  it("renders the Experience component correctly", () => {
    const { container, getByTestId, getByRole } = render(<Experience />);
    expect(container).toMatchSnapshot();

    const experienceContainer = getByTestId("experience-container");
    expect(experienceContainer).toBeInTheDocument();

    const educationTitle = getByRole("heading", { name: "Education Title" });
    expect(educationTitle).toBeInTheDocument();

    const workExperienceTitle = getByRole("heading", { name: "Work Experience Title" });
    expect(workExperienceTitle).toBeInTheDocument();

    const h3Elements = container.querySelectorAll("h3");
    expect(h3Elements.length).toEqual(2);
    expect(h3Elements[0].textContent).toBe("Test University");

    const paragraphElements = container.querySelectorAll("p");
    expect(paragraphElements.length).toEqual(8);

    expect(paragraphElements[0].textContent).toBe("Programme");
    expect(paragraphElements[1].textContent).toBe("2020 -> 2023");

    expect(h3Elements[1].textContent).toBe("Work Place");
    expect(paragraphElements[2].textContent).toBe("2020 -> 2023");
  });
});
