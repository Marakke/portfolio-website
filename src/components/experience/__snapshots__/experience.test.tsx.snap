// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experience renders the Experience component correctly 1`] = `
<div>
  <div
    class="experience-container"
    data-testid="experience-container"
  >
    <div
      class="education-container"
    >
      <h2>
        Education Title
      </h2>
      <div
        class="education-content"
      >
        <h3
          class="university"
        >
          Test University
        </h3>
        <div
          class="programmes"
        >
          <div
            class="programme"
          >
            <p
              class="name"
            >
              Programme
            </p>
            <p
              class="years"
            >
              2020
               
              -&gt;
               
              2023
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="work-experience-container"
    >
      <h2>
        Work Experience Title
      </h2>
      <div
        class="work-places"
      >
        <div
          class="work-place"
        >
          <h3
            class="name"
          >
            Work Place
          </h3>
          <p
            class="years"
          >
            2020
             
            -&gt;
             
            2023
          </p>
        </div>
      </div>
    </div>
    <div
      class="lines experience"
    >
      <p
        class="line experience"
      />
      <p
        class="line experience"
      />
      <p
        class="line experience"
      />
      <p
        class="line experience"
      />
      <p
        class="line experience"
      />
    </div>
  </div>
</div>
`;
